### ISSUE TITLE
Add total number of features & genes to check_fasta_gff?

### ISSUE THREAD

**Issue by FlicAnderson** [riboviz/riboviz#280]
Having recently used the newly refactored (#25) `check_fasta_gff` tool on the [example-datasets/#10 fly dataset] transcriptome files (https://github.com/riboviz/example-datasets/issues/10#issuecomment-782101703), I was wondering if a useful addition to the output might be the **total number of features in the .gff**.

Currently, the output is reported thus: 
```
# Created by: RiboViz
# Date: 2021-02-18 19:42:25.510660
# Command-line tool: /exports/eddie3_homes_local/fanders6/riboviz/riboviz/riboviz/tools/check_fasta_gff.py
# File: /exports/eddie3_homes_local/fanders6/riboviz/riboviz/riboviz/check_fasta_gff.py
# Version: commit 7e62c9f8da4f66f9c18aca30394f44fa4a7fa1ef date 2021-01-20 10:18:00-08:00
# fasta_file: A-Dm_2014/Dmel_filtered.fasta
# gff_file: A-Dm_2014/Dmel_filtered_UTRreplace.gff3
# start_codons: ['ATG']
# NoStopCodon: 6953
# NoStartCodon: 6948
# InternalStopCodon: 6596
# IncompleteFeature: 33
# SequenceNotInGFF: 0
# SequenceNotInFASTA: 0
# MultipleCDS: 0
# DuplicateFeatureId: 0
# NoIdName: 0
# DuplicateFeatureIds: 0
Sequence	Feature	Issue	Data
CG33282-RB	31165	InternalStopCodon	
CG10019-RD	31215	InternalStopCodon	
CG43707-RI	31219	InternalStopCodon	
lectin-28C-RB	31764	IncompleteFeature	
lectin-28C-RB	31764	NoStopCodon	AGN
lectin-28C-RB	31764	InternalStopCodon
# ...
# a bunch more features & such.
```

While I'm not an expert in estimating the "usual" amount of issues in a .gff, it definitely would be more intuitive if I knew what proportion of all the features had an issue reported here, so I could decide to either: 
a) bin the whole lot and start again after putting out the fire, or: 
b) whether it's just in need of some selective tweaking to improve, or, 
c)  if the transcriptome files are entirely fine to use as-is. 

Currently, I need to do a few extra steps to figure out the number of features held in the .gff file and then do some envelope maths to get a better idea what's happening.  Given that the code is already checking uniqueness of feature IDs etc, then this might not be too much of a stretch? 

@ewallace It'd be good to hear your thoughts on this, you've probably got a better feel for what you're looking for.

**Comment by FlicAnderson** [comment]
(Added to 'to review' so we can discuss at the dev meeting where to put this ticket)

**Comment by ewallace** [comment]
I think we should do this, adding both:
* total number of sequences in fasta file - should be easy as it's counting the number of keys in the dict of sequences
* total number of features of each kind in gff file - counting CDS features is easy so the right place to start

This should be just a couple of lines of code.

**Comment by mikej888** [comment]
Plus updates to `riboviz.test.test_check_fasta_gff`.

**Comment by mikej888** [comment]
Commit 8e3603b of [check-fasta-gff-total-features-genes-280](https://github.com/riboviz/riboviz/tree/check-fasta-gff-total-features-genes-280) adds support for printing number of sequences, number of features, number of CDS features, and adding this to the TSV file header also.

```console
$ python -m riboviz.tools.check_fasta_gff \
    -f data/yeast_CDS_w_250utrs.fa \
    -g data/yeast_CDS_w_250utrs.gff3 \
    -o check_data_CDS.tsv -v
...
Configuration:
fasta_file	data/yeast_CDS_w_250utrs.fa
gff_file	data/yeast_CDS_w_250utrs.gff3
start_codons	['ATG']

Metadata:
NumSequences	5812
NumFeatures	17436
NumCDSFeatures	5812

Issue summary:
Issue	Count
InternalStopCodon	17
NoStartCodon	1
DuplicateFeatureId	0
...

$ cat check_data_CDS.tsv
...
# fasta_file: data/yeast_CDS_w_250utrs.fa
# gff_file: data/yeast_CDS_w_250utrs.gff3
# start_codons: ['ATG']
# NumSequences: 5812
# NumFeatures: 17436
# NumCDSFeatures: 5812
# InternalStopCodon: 17
# NoStartCodon: 1
# DuplicateFeatureId: 0
...
Sequence	Feature	Issue	Data
Q0050	Q0050	InternalStopCodon	
Q0055	Q0055	InternalStopCodon	
Q0060	Q0060	InternalStopCodon	
...
```

**Comment by mikej888** [comment]
Created PR #299.

**Comment by mikej888** [comment]
Merged and closed #299.
